src/realm/query_expression.hpp

```diff
@@ -2598,6 +2768,11 @@ class Columns : public Subexpr2<T> {
         return m_sg ? get_column_base().get_column_index() : m_column_ndx;
     }
 
+    SizeOperator<Size<T>> size()
+    {
+        return SizeOperator<Size<T>>(this->clone(nullptr));
```

Comment:
Is there a reason to support `size()` on columns types besides strings, binary, links and subtables? This feels like it should be moved to the `Columns<StringData>` and `Columns<BinaryData>` specializations.